docker-compose.yml

```diff
@@ -76,71 +78,55 @@ services:
         # to disable persistence
         rm -f /data/dump.rdb && redis-server
       "
-    logging:
-      driver: none
     networks:
       default:
         aliases:
           - ${REDIS_HOST}
 
-  sqs:
-    image: localstack/localstack:0.11.5
-    environment:
-      - HOSTNAME_EXTERNAL=${SQS_HOST}
-      - IMAGE_NAME=localstack/localstack:0.11.5
-      - SERVICES=sqs:${SQS_PORT}
-    ports:
-      # Used to insert data in local grapl via upload-*.py
-      - 127.0.0.1:${SQS_PORT}:${SQS_PORT}
-    networks:
-      default:
-        aliases:
-          - ${SQS_HOST}
-    logging:
-      driver: none
-
-  secretsmanager:
-    image: localstack/localstack:0.11.5
-    environment:
-      - HOSTNAME_EXTERNAL=${SECRETSMANAGER_HOST}
-      - IMAGE_NAME=localstack/localstack:0.11.5
-      - PORT_WEB_UI=${SECRETSMANAGER_WEB_UI_PORT}
-      - SERVICES=secretsmanager:${SECRETSMANAGER_PORT}
-    ports:
-      - 127.0.0.1:${SECRETSMANAGER_WEB_UI_PORT}:${SECRETSMANAGER_WEB_UI_PORT}
-    networks:
-      default:
-        aliases:
-          - ${SECRETSMANAGER_HOST}
-    logging:
-      driver: none
-
+  # While localstack does offer S3 capabilities, it's not actually
+  # MinIO, which is a very full-featured and faithful S3 clone. We
+  # have noticed test failures when we use localstack's s3, so we will
+  # continue to use MinIO.
   s3:
     image: minio/minio
     command: server /data
     environment:
-      - MINIO_ACCESS_KEY=minioadmin
+      - MINIO_ACCESS_KEY=${S3_ACCESS_KEY_ID}
       - MINIO_REGION_NAME=${AWS_REGION}
-      - MINIO_SECRET_KEY=minioadmin
+      - MINIO_SECRET_KEY=${S3_ACCESS_KEY_SECRET}
     ports:
-      - 127.0.0.1:${S3_PORT}:${S3_PORT}
+      - 127.0.0.1:${S3_PORT_EXTERNAL}:${S3_PORT_INTERNAL}
     networks:
       default:
         aliases:
           - ${S3_HOST}
-    logging:
-      driver: none
 
-  dynamodb:
-    image: amazon/dynamodb-local
+  localstack:
+    image: localstack/localstack:0.12.6
+    ports:
+      # We'll expose localstack's edge port for ease of use with
+      # things like the AWS CLI, Pulumi, etc. We'll also re-expose it
+      # for various other services that we might want to otherwise
+      # target explicitly from our workstations.
+      #
+      # You can just stick with the main localstack port, though, and
+      # should get proxied through to where you're trying to go.
+      - 127.0.0.1:${LOCALSTACK_PORT}:${LOCALSTACK_PORT}
+      # As such, these other port exposures can be considered "legacy"
+      # at this point.
+      - 127.0.0.1:${SQS_PORT_EXTERNAL}:${LOCALSTACK_PORT}
```

Comment:
given this comment - can we remove this second port forward then?

Reply:
If nobody is using it, sure. I'll verify that today.

Reply:
It's not required, so we can remove it entirely.